docs/reference/contributing/target/rtc.md

```diff
@@ -1,6 +1,14 @@
 ### RTC
 
-The RTC HAL API provides a low-level interface to the Real Time Counter (RTC) of a target.
+Implementing RTC enables Mbed OS to keep track of the current time. Standard library time keeping functions, such as `time`, use it.
+
+#### Deprecation warning
+
+**All of the following information is intended for the new RTC HAL API, available on a feature branch, but not yet part of the master branch. We are planning to replace the API on master in time for the Mbed OS 5.9 release. If you intend for your port to be part of Mbed OS before May 2019, you need to implement both of the APIs. In this case, you need to submit your main port as a pull request against master, implementing the old API. Additionally, you need to raise another pull request against `feature-hal-spec-rtc` branch, implementing the new API described below.**
```

Comment:
Let's talk about this in person. I have a lot of thoughts and questions about this warning and this wording, and it would be faster to chat in person.